metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1198,15 +1198,20 @@ class MetalsLspService(
   }
 
   def didChangeWatchedFiles(
-      paths: List[AbsolutePath]
+      events: List[FileEvent]
   ): Future[Unit] = {
-    val importantPath =
-      paths
-        .filterNot(path =>
-          savedFiles.isRecentlyActive(path) || path.isDirectory
+    val importantEvents: List[FileWatcherEvent] =
+      events
+        .collect(event =>
+          event.getUri().toAbsolutePathSafe match {
+            case Some(path)
```

Comment:
Either this pattern match is non-exhaustive or I'm about to learn something new, which one is this? :D

Reply:
It's a partial function that way (I think):
https://scastie.scala-lang.org/QTTHAPKfSWmjqX3CA0GsTQ